main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import "fmt"
+
+func worker(id int, jobs <-chan int, results chan<- int) {
+	for j := range jobs {
+		go func() {
```

Comment:
Passing the value of `j` from the loop to the go func is a common error, because this will take the last value that j contains for every concurrent method

one solution can be something like this 
```
for j := range jobs {
  jVal := j
    go func(){
         switch iVal % 3 {
...
```